src/java/org/apache/cassandra/index/sai/view/View.java

```diff
@@ -61,6 +68,12 @@ public View(IndexContext context, Collection<Descriptor> sstables, Collection<SS
         List<Interval<Key, SSTableIndex>> keyIntervals = new ArrayList<>();
         for (SSTableIndex sstableIndex : indexes)
         {
+            if (!sstableIndex.reference())
+            {
+                this.view.values().forEach(SSTableIndex::release);
+                throw new IllegalStateException("Failed to reference the index " + sstableIndex);
```

Comment:
I am not sure the reason of referencing `SSTableIndex` and fail inside `View#constructor`. If there are concurrent `IndexViewManager#update`, one of them may throw `IllegalStateException`.

Reply:
If I understand correctly, the goal is to make `View.java` a referenced view, so SAI query can use it without referencing. If that's the case, we can reference `SSTableIndex` out side of `View#constructor`, e.g. in `IndexViewManager#update` and retry if some `SSTableIndex` failed to reference.